src/main/java/seedu/clinkedin/model/person/Rating.java

```diff
@@ -0,0 +1,53 @@
+package seedu.clinkedin.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.clinkedin.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's rating in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidRatingStr(String)}
+ */
+public class Rating {
+
+    public static final String MESSAGE_CONSTRAINTS = "Rating should only be an integer between 1 to 10 inclusive.";
+    public static final String VALIDATION_REGEX = "/^([0-9]|10)$/";
+    public final int value;
+
+    /**
+     * Constructs a {@code Rating}.
+     *
+     * @param ratingStr A valid rating number.
+     */
+    public Rating(String ratingStr) {
+        requireNonNull(ratingStr);
+        checkArgument(isValidRatingStr(ratingStr), MESSAGE_CONSTRAINTS);
+        int rating = Integer.parseInt(ratingStr);
+        value = rating;
+    }
+
+    /**
+     * Returns true if a given string can be parsed into an integer and is between 1 and 10 inclusive.
+     */
+    public static boolean isValidRatingStr(String test) {
+        return test.equals("1") || test.equals("2") || test.equals("3") || test.equals("4") || test.equals("5")
```

Comment:
Maybe Integer.parseInt(test.trim()) then checking the value would be better? This is very unclean >.<

Reply:
I totally understand where you are coming from! But this only works if the String is an integer. If I do Integer.parseInt("4.9") it will just throw exception, and I guess I can handle exceptions (which could be done, just that I read that having exceptions as part of logic might not be best practice haha)